src/bin/cargo/cli.rs

```diff
@@ -595,6 +606,11 @@ See '<cyan,bold>cargo help</> <cyan><<command>></>' for more information on a sp
                 .value_name("WHEN")
                 .global(true),
         )
+        .arg(
+            opt("warnings", "Warning behavior")
```

Comment:
> Since we have --config "term.warnings='error'" do we need the --warnings CLI option?

Reply:
I lean towards
- Stabilize the config and add the flag later (take the incremental approach).
  - That also means that we'll mostly have early adopters at first which will reduce the amount annoyance if we say skip cargo warnings and add them later
  - This will help us vet how much the flag is needed compared to the config
- When we add the flag, maybe only put it on warning-heavy commands (`cargo check`, `cargo build`) to help people discover it without flooding every command with it (`cargo version --warnings error`?)